lib/components/user/after-signin-switch.js

```diff
@@ -0,0 +1,44 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+
+import { isNewUser } from '../../actions/user'
+
+import AfterSignInScreen from './after-signin-screen'
+import withLoggedInUserSupport from './with-logged-in-user-support'
+
+/**
+ * This screen is flashed just after user sign in while state.user.loggedInUser is fetched.
+ * Once state.user.loggedInUser is available:
+ * - For new users, route them to the account page (it will show account setup).
+ * - For existing users, route them to the itinerary search that was in place before login.
+ */
+class AfterSignInSwitch extends Component {
+  componentDidUpdate () {
+    const { loggedInUser } = this.props
+
+    if (loggedInUser) {
+      const nextUrl = isNewUser(loggedInUser) ? '/account' : '/'
+      // Prevent the current URL from showing in the browser history.
+      window.location.replace(window.location.href.replace('#/signedin', `#${nextUrl}`))
```

Comment:
This component probably shouldn't exist and any logic related to changing the URL should be put into an action.

Reply:
And redux actions are triggered from mounted components?

Reply:
Changed the routing to use a modified version of `routeTo`.